vllm/compilation/backends.py

```diff
@@ -238,7 +239,7 @@ def compile(
         assert compiled_graph is not None, "Failed to compile the graph"
 
         # store the artifact in the cache
-        if not envs.VLLM_DISABLE_COMPILE_CACHE and handle is not None:
+        if is_compile_cache_enabled() and handle is not None:
```

Comment:
The user might also pass `force_disable_caches` as part of `CompilationConfig.inductor_compile_config`, we should check that as well

Reply:
Good catch, added a check for compilation config in the `is_compile_cache_enabled`